src/expr.rs

```diff
@@ -221,38 +222,108 @@ impl<'a> FmtVisitor<'a> {
         format!("({})", item_str)
     }
 
+    fn rewrite_binary_op(&mut self,
+                         op: &ast::BinOp,
+                         lhs: &ast::Expr,
+                         rhs: &ast::Expr,
+                         width: usize,
+                         offset: usize)
+        -> String
+    {
+        let operator_str = self.snippet(op.span);
+
+        // 1 = space between lhs expr and operator
+        let mut result = self.rewrite_expr(lhs, width - 1 - operator_str.len(), offset);
+
+        result.push(' ');
+        result.push_str(&operator_str);
+
+        let remaining_width = match result.rfind('\n') {
+            Some(idx) => {
+                let space = config!(max_width) + idx;
+                let used = offset + result.len();
```

Comment:
I don't think the maths here is correct. I think in this case the remaining width is `config!(max_width) - (result.len() - idx)`. I guess there needs to be an overflow check in there.